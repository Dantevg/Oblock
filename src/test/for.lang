{
	iterable := {
		i := 0
		iterate := () => {
			i = i+1
			return i < 5 && i || nil
		}
	}
	
	for x in iterable: print x
}