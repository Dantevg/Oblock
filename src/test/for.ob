iterable = {
	iterate = () => {
		var i = 0
		return () => {
			i += 1
			return i < 5 and i or nil
		}
	}
}

for x in iterable: print x

list = [10, y = 20, 30]
for x in list: print x

for x in [1,2,3,4]: print x
