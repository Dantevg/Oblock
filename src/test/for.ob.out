1
2
3
4
10
30
1
2
3
4
